unlock-protocol.com/blog/unlocking-smart-contracts.md

```diff
@@ -0,0 +1,145 @@
+  }
+
+  // You can also have free features of course
+}
+```
+
+Pretty straight forward, right?  Let’s step through it.
+
+First we need to install an NPM package with this command:
+
+```
+npm i hardlydifficult-ethereum-contracts
+```
+
+Now we can review the code...
+
+```
+import 'hardlydifficult-ethereum-contracts/contracts/interfaces/IPublicLock.sol';
+```
+
+Importing the interface allows us to make calls to the Lock contract, given its address.  
+
+
+```
+IPublicLock public lock;
+```
+
+To gain access, users must purchase a key to the Lock for this content.  This stores the address for the Lock contract.  By using `IPublicLock` instead of `address` we simplify making calls within the contract.  `public` allows end-users to read the Lock’s address, confirming they are purchasing the correct Key.
+
+
+```
+constructor(IPublicLock _lockAddress) public
+{
+  lock = _lockAddress;
+}
+```
+
+We have to set the address for the Lock to use at some point.  If you know it when this contract is deployed you can simply assign it in the constructor. However you might also consider a privileged call instead, allowing you to update the Lock in the future if needed.  Here’s an example of a [MutableLock](https://github.com/HardlyDifficult/unlock-example-contracts/blob/master/contracts/MutableLock.sol).
+
+```
+require(lock.getHasValidKey(msg.sender), 'Purchase a key first!');
+```
+
+`getHasValidKey` checks if the given address owns a valid (unexpired) key and returns true or false.  We place this in a `require` statement so that the entire transaction fails if they do not.  If you prefer, this could be moved to a `modifier` instead.  Alternatively you could use an `if` condition instead to modify behavior for Key owners somehow (we have a few ideas about that below).
+
+That’s the contract implementation!  The hardest part can be testing though.  We need a way to run your contract locally, and to make calls with and without owning a Key.  For this we have a script to get you started:
+
+```javascript
+const { protocols } = require("hardlydifficult-ethereum-contracts");
+const PaidOnlyFeature = artifacts.require("PaidOnlyFeature");
+
+contract("PaidOnlyFeature", accounts => {
+  let lock;
+  let featureContract;
+  const keyOwner = accounts[3];
+
+  beforeEach(async () => {
+    lock = await protocols.unlock.createTestLock(
+      web3,
+      accounts[9], // Unlock Protocol owner
+      accounts[1], // Lock owner
+      {
+        keyPrice: web3.utils.toWei("0.01", "ether")
+      }
+    );
+
+    featureContract = await PaidOnlyFeature.new(lock.address);
+
+    await lock.purchaseFor(keyOwner, {
+      from: keyOwner,
+      value: await lock.keyPrice()
+    });
+  });
+
+  it("Key owner can call the function", async () => {
+    await featureContract.paidOnlyFeature({
+      from: keyOwner
+    });
+  });
+});
+```
+
+This line does all the heavy lifting for you:
+
+```javascript
+lock = await protocols.unlock.createTestLock(
```

Comment:
@julien51 do we want to replace this with calls to unlock-js instead?  I haven't looked into deploying with that yet but I suspect it wouldn't be too hard.  If we can change this and publish the interface than we can drop mention of my package here.